cmd/worker.go

```diff
@@ -68,26 +71,34 @@ func worker(ctx context.Context, cfg *config.AppConfig) {
 
 	engine := query.NewEngine("infratographer", spiceClient, query.WithPolicy(policy), query.WithLogger(logger))
 
-	subscriber, err := pubsub.NewSubscriber(ctx, cfg.Events.Subscriber, engine, pubsub.WithLogger(logger))
+	events, err := events.NewConnection(cfg.Events.Config, events.WithLogger(logger))
+	if err != nil {
+		logger.Fatalw("failed to initialize events", "error", err)
+	}
+
+	subscriber, err := pubsub.NewSubscriber(ctx, events, engine,
+		pubsub.WithLogger(logger),
+	)
 	if err != nil {
 		logger.Fatalw("unable to initialize subscriber", "error", err)
 	}
 
-	defer subscriber.Close()
+	topics := viper.GetStringSlice("events.topics")
```

Comment:
Should we make this a config item instead of just a Viper call?

Reply:
yep, good call.